resteasy-jaxrs/src/main/java/org/jboss/resteasy/spi/WeldUtil.java

```diff
@@ -0,0 +1,29 @@
+package org.jboss.resteasy.spi;
+
+class WeldUtil
+{
+   private static final String WELD_PROXY_INTERFACE_NAME = "org.jboss.weld.bean.proxy.ProxyObject";
+
+   /**
+    * Whether the given class is a proxy created by Weld or not. This is
+    * the case if the given class implements the interface
+    * {@code org.jboss.weld.bean.proxy.ProxyObject}.
+    * 
+    * This is needed because of https://issues.jboss.org/browse/WELD-1539 and
+    * https://issues.jboss.org/browse/WELD-1914 which are unsolved.
+    *
+    * @param clazz the class of interest
+    *
+    * @return {@code true} if the given class is a Weld proxy,
+    * {@code false} otherwise
+    */
+   static boolean isWeldProxy(Class<?> clazz) {
+      for ( Class<?> implementedInterface : clazz.getInterfaces() ) {
```

Comment:
You could also easily filter out classes which are not synthetic, i.e. those for which `Class.isSynthetic()` returns false  - see also https://issues.jboss.org/browse/WELD-1914?focusedCommentId=13099225&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-13099225.